cluster-autoscaler/cloudprovider/aws/auto_scaling.go

```diff
@@ -36,12 +41,55 @@ type autoScaling interface {
 // autoScalingWrapper provides several utility methods over the auto-scaling service provided by AWS SDK
 type autoScalingWrapper struct {
 	autoScaling
-	launchConfigurationInstanceTypeCache map[string]string
+	launchConfigurationInstanceTypeCache *expirationStore
+}
+
+// expirationStore cache the launch configuration with their instance type.
+// The store expires its keys based on a TTL. This TTL can have a jitter applied to it.
+// This allows to get a better repartition of the AWS queries.
+type expirationStore struct {
+	cache.Store
+	jitterClock *jitterClock
+}
+
+type instanceTypeCachedObject struct {
+	name         string
+	instanceType string
+}
+
+type jitterClock struct {
+	clock.Clock
+
+	jitter bool
+	sync.RWMutex
+}
+
+func newLaunchConfigurationInstanceTypeCache() *expirationStore {
+	jc := &jitterClock{}
+	return &expirationStore{
+		cache.NewExpirationStore(func(obj interface{}) (s string, e error) {
+			return obj.(instanceTypeCachedObject).name, nil
+		}, &cache.TTLPolicy{
+			TTL:   time.Minute * 20,
```

Comment:
I am thinking if 20 minutes work for most of the users. Not sure if user will change it if we add an command option for it.  Let's at least have const.

Reply:
As you want, I've done the needed changes 👍